app/scripts/controllers/pod.js

```diff
@@ -87,6 +88,54 @@ angular.module('openshiftConsole')
       }
     };
 
+    var calculateCharacterBoundingBox = function() {
+      var calcSpan = $("<span>")
+      .css({
+        position: "absolute",
+        top: "-100px"
+      })
+      .addClass("terminal-font")
+      .text("xxxxxxxxxx")
+      .appendTo($('body'));
+      var vals = {
+        width: calcSpan.width() / 10, // average across several characters to take into account internal spacing
+        height: calcSpan.height()
+      };
+      calcSpan.remove();
+      return vals;
+    };
+
+    var characterBoundingBox = calculateCharacterBoundingBox();
+    var win = $( window );
+    var calculateTerminalSize = function(){
+      if (!characterBoundingBox.height || !characterBoundingBox.width) {
+        return;
+      }
+      $scope.$apply(function() {
+        var terminalWrapper = $('.container-terminal-wrapper').get(0);
+        var r = terminalWrapper.getBoundingClientRect();
+        var windowWidth = win.width();
+        var windowHeight = win.height();
+        var termWidth = windowWidth - r.left - 40;
+        var termHeight = windowHeight - r.top - 50;
+        $scope.terminalCols = Math.max(_.floor(termWidth / characterBoundingBox.width), 80);
+        $scope.terminalRows = Math.max(_.floor(termHeight / characterBoundingBox.height), 24);
+      });
+    };
+
+    if (!characterBoundingBox.height || !characterBoundingBox.width) {
+      Logger.warn("Unable to calculate the bounding box for a character.  Terminal will not be able to resize.");
+    }
+    else {   
+      $(window).on('resize.terminalsize', _.debounce(calculateTerminalSize, 100));
+    }
+
+    $scope.$watch('selectedTab.terminal', function(newValue) {
+      if (newValue === true) {
```

Comment:
Is `newValue` ever a truthy value other than boolean `true`? Otherwise I'd just use `if (newValue) { ... }`.

Code reads better with a name like `terminalTabSelected`